code/Weapons/Weapon.cs

```diff
@@ -150,6 +150,38 @@ public bool HasAmmo()
 		return Ammo != 0;
 	}
 
+	public Vector3 GetStartPosition( bool isDroppable = false )
+	{
+		if ( isDroppable )
+			return Position.WithY( 0 );
+
+		var muzzle = GetAttachment( "muzzle" );
+		if ( muzzle is null )
+			return Grub.EyePosition;
+
+		var tr = Trace.Ray( Grub.Controller.Hull.Center + Grub.Position, muzzle.Value.Position + muzzle.Value.Rotation.Forward )

```

Comment:
Got any better ideas than `Grub.Controller.Hull.Center + Grub.Position`?

Reply:
Nope, that is straightforward and I think it is fine :)